airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/report_streams/products_report.py

```diff
@@ -275,6 +284,27 @@ def _download_report(self, report_info: ReportInfo, url: str) -> List[dict]:
         """
         return super()._download_report(None, url)
 
+    @backoff.on_exception(
+        backoff.expo,
+        (
+            requests.exceptions.Timeout,
+            requests.exceptions.ConnectionError,
+            TooManyRequests,
+        ),
+        max_tries=10,
+    )
+    def _send_http_request(self, url: str, profile_id: int, json: dict = None):
```

Comment:
> Should this change apply on the ReportStream level and not on the SponsoredProductsReportStream one?

Other reports don't fail if Auth header were provided for download url. So it's an issue only for sponsored_products_report stream. 

> If this is not on the ReportStream, should we pass the proper parameters to re-use the logic (retrying, exceptions, etc...)?

You mean handle 400 error in case we face with it again? It's a good point as error handling in the source is unreliable: [example](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/report_streams/report_streams.py#L429). We discussed this with @girarda,  the source is planned to be migrated to low code, so as part of this work we can refactor error handling.

Reply:
> Other reports don't fail if Auth header were provided for download url. So it's an issue only for sponsored_products_report stream.

Doesn't that seem odd? 

> You mean handle 400 error in case we face with it again?

I mean, sharing the logic with the same method. Something like:
```
class ReportStream(BasicAmazonAdsStream, ABC):
    def __init__(<...>):
        self._session = requests.Session()
        self._session.auth = authenticator
        
        self._report_download_session = self._session
        
    def _send_http_request(self, url: str, profile_id: int, json: dict = None, is_download_report=False):
        headers = self._get_auth_headers(profile_id)
        if json:
            response = self._session.post(url, headers=headers, json=json)
        else:
            session = self._report_download_session if is_download_report else self._session
            response = session.get(url, headers=headers)
        if response.status_code == HTTPStatus.TOO_MANY_REQUESTS:
            raise TooManyRequests()
        return response
        
class SponsoredProductsReportStream(ReportStream):
    def __init__(<...>):
        self._report_download_session = requests.Session()
```

This way, if we update how we perform HTTP requests in `ReportStream`, the dev does not need to know that `SponsoredProductsReportStream` re-implement this method and therefore needs to be updated as well

Reply:
> Doesn't that seem odd?

I know it looks weird, but my testing showed this behavior.  I think it's because of amazon ads stores this kind of report(sponsored_products_report_stream) in s3.amazonaws, so download urls, for example, for `sponsored_brands_report_stream` and `sponsored_products_report_stream` are different, so we can't use same logic for these reports. 

You can see in the affected workspaces for oc issue, that data for sponsored_brands_report_stream was downloaded as expected, but they got 400 status for sponsored_products_report_stream download.